### DISCUSSION TITLE
How is Vanara compared to terrafx.interop.windows?

### DISCUSSION THREAD

**Discussion by ghost** [dahall/Vanara#433] [Q&A] [answered]
https://github.com/terrafx/terrafx.interop.windows

**Comment by dahall** [comment, marked as answer]
Terrafx is a great and complete library very similar in function to Vanara. The major difference is that Terrafx takes a direct approach using unsafe methods and structures and a near identical expression. Vanara takes a .NET centric approach that attempts to use managed patterns and base class elements. For example, Terrafx might take a complex structure and provide an unsafe pointer to an unsafe structure and leave string or other marshaling activities to the user. Vanara takes that same method and provides a method overload that custom marshals the structure directly into its managed form.
